libraries/AP_FETtecOneWire/AP_FETtecOneWire.cpp

```diff
@@ -0,0 +1,759 @@
+    _response_length[OW_REQ_TYPE] = 1;
+    _response_length[OW_REQ_SN] = 12;
+    _response_length[OW_REQ_SW_VER] = 2;
+    _response_length[OW_SET_FAST_COM_LENGTH] = 1;
+    _response_length[OW_SET_TLM_TYPE] = 1;
+}
+
+/**
+  initialize the serial port
+*/
+void AP_FETtecOneWire::init()
+{
+    AP_SerialManager& serial_manager = AP::serialmanager();
+    _uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_FETtecOneWire, 0);
+    if (_uart) {
+        _uart->set_flow_control(AP_HAL::UARTDriver::FLOW_CONTROL_DISABLE);
+        _uart->set_unbuffered_writes(true);
+        _uart->set_blocking_writes(false);
+        _uart->begin(BAUDRATE);
+        _uart_initialised = true;
+    }
+}
+
+/**
+  update configuration periodically to accommodate for parameter changes and test if the current configuration is OK
+*/
+void AP_FETtecOneWire::configuration_update()
+{
+    if (!hal.util->get_soft_armed()) { // configuration updates are only done when vehicle is disarmed
+
+        const uint32_t now = AP_HAL::millis();
+        if ((now - _last_config_update_ms > 2000) || _last_config_update_ms == 0) {  // only runs once every 2 seconds
+            _last_config_update_ms = now;
+
+            if (!_uart->is_dma_enabled()) {
+                GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "FTW UART needs DMA");
+                return; // will not work at all without DMA
+            }
+
+            // get the user-configured FETtec ESCs bitmask parameter
+            _mask = uint16_t(_motor_mask.get());
+            // tell SRV_Channels about ESC capabilities
+            SRV_Channels::set_digital_outputs(_mask, 0);
+
+            uint16_t smask = _mask;
+            _nr_escs_in_bitmask = 0;
+
+            // count the number of user-configured FETtec ESCs in the bitmask parameter
+            for (uint8_t i = 0; i < MOTOR_COUNT_MAX; i++) {
+                SRV_Channel* c = SRV_Channels::srv_channel(i);
+                if (c == nullptr || (smask & 0x01) == 0x00) {
+                    break;
+                }
+                smask >>= 1;
+                _nr_escs_in_bitmask++;
+            }
+
+#if HAL_WITH_ESC_TELEM
+            _update_rate_hz = AP::scheduler().get_loop_rate_hz();
+            _crc_error_rate_factor = 100.0f/(float)_update_rate_hz; //to save the division in loop, precalculate by the motor loops 100%/400Hz
+            // TLM recovery, if e.g. a power loss occurred but FC is still powered by USB.
+            const uint8_t num_active_escs = AP::esc_telem().get_num_active_escs();
+            if (num_active_escs < _nr_escs_in_bitmask) {
+                if ((now -_last_scan_ms > 5000) || _last_scan_ms == 0) {
+                    GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "FTW found only %i of %i ESCs", num_active_escs, _nr_escs_in_bitmask);
+                    _last_scan_ms = now;
+                    _scan_active = 0;
+                    _setup_active = 0;
+                    _set_full_telemetry_active = 0;

```

Comment:
```suggestion
                    _set_full_telemetry_active = 1;
```
@teezeefpv  Shouldn't this be inited to 1?

Reply:
Yes correct.